### PR TITLE
[core] Render config errors

### PR DESCRIPTION
**Pull request by jsotuyod**
Before submitting a PR, please check that:
 - [x] The PR is submitted against `master`. The PMD team will merge back to support branches as needed.
 - [x] `mvn test` passes.
 - [x] `mvn checkstyle:check` passes. [Check this for more info](https://github.com/pmd/pmd/blob/master/CONTRIBUTING.md#code-style)

**PR Description:**

- All renderers previously including processing errors now also include configuration errors.
- I take the chance to fix a couple of issues with `VBHTMLRenderer`, that was rendering processing errors using `toString()` (which has a default implementation)
- Notice that renderers that didn't previously include processing errors are unmodified
- This resolves part 1 of #194

### CODE REVIEW

File: pmd-core/src/main/java/net/sourceforge/pmd/renderers/XMLRenderer.java [hunk 1]
```diff
@@ -152,6 +152,17 @@ public void end() throws IOException {
                 writer.write(buf.toString());
             }
         }
+        
+        // config errors
+        for (final Report.ConfigurationError ce : configErrors) {
+            buf.setLength(0);
+            buf.append("<configerror ").append("rule=\"");
```
Comment (adangel) [inline pmd-core/src/main/java/net/sourceforge/pmd/renderers/XMLRenderer.java]: I'm wondering whether we have documented the XML report format somewhere.... We are adding now a new element, which might break other tools. I think maven-pmd-plugin is safe, since it doesn't rely on that renderer, but maybe the Jenkins plugin does?

Comment (jsotuyod) [inline pmd-core/src/main/java/net/sourceforge/pmd/renderers/XMLRenderer.java]: yes, tools such as jenkins are possibly going to break, but that's why this change is ok for 6.0.0.

We should probably create an XML Schema for this report and expose it...

Comment (adangel) [inline pmd-core/src/main/java/net/sourceforge/pmd/renderers/XMLRenderer.java]: Yes, I'm fine with breaking compatibility. We just need to be aware of that.
Creating a schema probably makes sense.

Comment (jsotuyod) [inline pmd-core/src/main/java/net/sourceforge/pmd/renderers/XMLRenderer.java]: I did include a notice in the release changes, but feel free to edit it if you don't consider it explicit enough: https://github.com/pmd/pmd/pull/519/files#diff-b7709f07439de5204938d714fe93ed84

I'll get to the XML Schema on a separate PR / task

**Comment by adangel** [comment]
Merged! Thanks again!
